include/bls12-381/fp.hpp

```diff
@@ -48,7 +48,12 @@ class fp
     bool isLexicographicallyLargest() const;
     template<size_t N> static fp modPrime(const std::array<uint64_t, N>& k);
 
-    auto operator<=>(const fp&) const = default;
+    std::strong_ordering operator<=>(const fp& e) const {
+        constexpr std::strong_ordering res[] = { std::strong_ordering::less,
+                                                 std::strong_ordering::equal,
+                                                 std::strong_ordering::greater };
+        return res[cmp(e) + 1];
```

Comment:
if function `cmp` will be rewritten the way that it will return out of range value then this code will crash.
following code looks safer to me.
```suggestion
        switch (cmp(e)) {
        case -1:
        return std::strong_ordering::less;
        case 0:
        return std::strong_ordering::equal;
        case 1:
        return std::strong_ordering::greater;
        }
        
        // big bang is happening here. can we throw or assert?
        return -1;
```

Reply:
Well, this alternate version is less efficient, and there is no reason to expect that `cmp` would be rewritten to return something else than -1, 0 or 1.

Reply:
@dimas1185 I have changed `scalar::cmp` to also return `std::strong_ordering`, so no awkward conversion anymore.